test/multipart.test.js

```diff
@@ -538,3 +538,81 @@ test('should also work with multipartIterator', function (t) {
     }
   })
 })
+
+test('should receive all field', function (t) {
+  t.plan(11)
+
+  const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+
+  fastify.register(multipart)
+
+  fastify.post('/', async function (req, reply) {
+    const recvField = {
+      upload: false,
+      hello: false,
+      willbe: false
+    }
+
+    for await (const part of req.parts()) {
+      recvField[part.fieldname] = true
+      if (part.file) {
+        t.equal(part.fieldname, 'upload')
+        t.equal(part.filename, 'README.md')
+        t.equal(part.encoding, '7bit')
+        t.equal(part.mimetype, 'text/markdown')
+        t.ok(part.fields.upload)
+
+        const original = fs.readFileSync(filePath, 'utf8')
+        await pump(
+          part.file,
+          concat(function (buf) {
+            t.equal(buf.toString(), original)
+          })
+        )
+        // wait for 1s to mimic situation that user side takes a long time to process every part
+        await new Promise((resolve, reject) => {
+          setTimeout(resolve, 1000)
```

Comment:
Please avoid using timers in tests.

Reply:
Then what's the standard way to simulate a time-consuming async action ?

Reply:
It's dependent on each case. I think just using a setImmediate would work in this case.